src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,64 @@
 package mate.academy;
 
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private static final int THREAD_POOL_SIZE = 1;
+    private static final int SHUTDOWN_TIMEOUT_MINUTES = 2;
+    private final ThreadPoolExecutor executor;
+    private final List<EventListener> listeners;
 
+    public EventManager() {
+        this.executor = new ThreadPoolExecutor(
+                THREAD_POOL_SIZE,
+                THREAD_POOL_SIZE,
+                0L,
+                TimeUnit.MILLISECONDS,
+                new LinkedBlockingQueue<>()
+        );
+        this.listeners = new CopyOnWriteArrayList<>();
     }
 
-    public void deregisterListener(EventListener listener) {
+    public void registerListener(EventListener listener) {
+        listeners.add(listener);
+        setUpNewThreadPoolSize();
+    }
```

Comment:
you can also prohibit addition of new listeners if we have called `shutdown()` and clear the list of listeners after that, but that is up to you